sw/device/lib/crypto/include/aes.h

```diff
@@ -95,44 +95,52 @@ typedef enum aes_padding {
 typedef struct gcm_ghash_context gcm_ghash_context_t;
 
 /**
- * Performs the AES initialization operation.
- *
- * The `aes_mode` and `aes_operation` are selected and the `key`, `iv`
- * are loaded into the register.
- *
- * @param key Pointer to the blinded key struct with key shares
- * @param iv Initialization vector, used for CBC, CFB, OFB, CTR modes
- * @param aes_mode Required AES mode of operation
- * @param aes_operation Required AES operation (encrypt or decrypt)
- * @return The result of the init operation
+ * Generates a new AES key.
+ *
+ * The caller should allocate and partially populate the blinded
+ * key struct, including populating the key configuration and
+ * allocating space for the keyblob. The caller should indicate
+ * the length of the allocated keyblob; this function will
+ * return an error if the keyblob length does not match
+ * expectations. For hardware-backed keys, the keyblob length is 0
+ * and the keyblob pointer may be NULL. For non-hardware-backed keys,
+ * the keyblob should be twice the length of the key. The value in
+ * the `checksum` field of the blinded key struct will be populated
+ * by the key generation function.
+ *
+ * @param key Destination blinded key struct
+ * @return The result of the cipher operation
```

Comment:
```suggestion
 * @return The result of the cipher operation.
```
The same for the other arguments.